sycl/doc/Array_Kernel_Parameters.md

```diff
@@ -0,0 +1,435 @@
+<h2>Array Parameters of SYCL Kernels</h2>
+
+<h3>Introduction</h3>
+
+This document describes the changes to support passing arrays to SYCL kernels 
+and special treatment of Accessor arrays.
+The following cases are handled:
```

Comment:
What happens if user code does fit into the supported cases?

Reply:
An compiler diagnostic is issued.